File: backend/src/main/java/acalidonio/bodegamovilbackend/domain/dto/response/ApiErrorResponse.java
package acalidonio.bodegamovilbackend.domain.dto.response;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class ApiErrorResponse {
    private int status;
    private Object message;
    private LocalDateTime time;
    private String uri;
}

File: app/src/main/java/com/acalidonio/bodegamovil/data/remote/dto/ApiErrorResponse.kt
package com.acalidonio.bodegamovil.data.remote.dto

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.JsonObject

@Serializable
data class ApiErrorResponse(
    val status: Int,
    val message: JsonElement,
    val time: String,
    val uri: String
) {
    fun getReadableMessage(): String {
        return try {
            if (message is JsonObject) {
                message.values.joinToString("\n") { it.toString().replace("\"", "") }
            } else {
                message.toString().replace("\"", "")
            }
        } catch (_: Exception) {
            "Ocurrió un error inesperado en el servidor."
        }
    }
}
